lib/puppet/util/log/destinations.rb

```diff
@@ -111,6 +112,8 @@ def handle(msg)
     if @json > 0
       @json > 1 ? @file.puts(',') : @json = 2
       @file.puts(Puppet::Util::Json.dump(msg.to_structured_hash))
+    elsif @jsonl
+      @file.puts(Puppet::Util::Json.dump(msg.to_structured_hash))
```

Comment:
Could you add a unit test, something like https://github.com/puppetlabs/puppet/blob/master/spec/unit/util/log/destinations_spec.rb#L82-L94, but with a file with `.jsonl` extension?

Reply:
D'oh, I failed to find the JSON test.  I just added a commit to check for JSON lines as well.  Thanks for the pointer to the test!